### ISSUE TITLE
File not found problem with model file

### ISSUE THREAD

**Issue by joecheriross** [gregdurrett/berkeley-entity#5]
Hi Greg,
I compiled the source files again and created berkeley-entity jar in target directory. Jar creation is successful. But while running the system with this jar I am getting the following error. Please help. 

ERROR: java.lang.RuntimeException: Can't write to models/cached/corefpruner-onto.ser.gz:
edu.berkeley.nlp.entity.GUtil$.save(GUtil.scala:28)
edu.berkeley.nlp.entity.coref.CorefPruner$.trainAndSaveKFoldModels(CorefPruner.scala:103)
edu.berkeley.nlp.entity.coref.CorefPruner$.buildPruner(CorefPruner.scala:93)
edu.berkeley.nlp.entity.coref.CorefSystem$.runTrain(CorefSystem.scala:140)
edu.berkeley.nlp.entity.coref.CorefSystem$.runTrainPredict(CorefSystem.scala:108)
edu.berkeley.nlp.entity.coref.CorefSystem.runTrainPredict(CorefSystem.scala)
edu.berkeley.nlp.entity.Driver.run(Driver.java:343)
edu.berkeley.nlp.futile.fig.exec.Execution.runWithObjArray(Execution.java:479)
edu.berkeley.nlp.futile.fig.exec.Execution.run(Execution.java:432)
edu.berkeley.nlp.entity.Driver.main(Driver.java:319)

Thanks,
Joe

Command: java -Xmx8g -jar target/scala-2.11/berkeley-entity-assembly-1.jar ++config/base.conf -execDir scratch -mode COREF_TRAIN_PREDICT  -testPath /tmp/test_input/  -docSuffix auto_conll -trainPath ./small_train/ -modelPath "models/joint-onto.ser.gz" -wikipediaPath "models/wiki-db-onto.ser.gz"  -useGoldMentions -pruningStrategy build:models/cached/corefpruner-onto.ser.gz:-5:5 -nerPruningStrategy build:models/cached/nerpruner-onto.ser.gz:-9:5 -outputPath /tmp/test_output/

**Comment by joecheriross** [comment]
Sorry. This issue is with existing jar as well. This is some other problem. The existing jar was working fine. Let me try to resolve it by my own. But in case you know possible causes let me know. 

Thanks,
Joe

**Comment by gregdurrett** [comment]
Does that directory exist? GUtil is trying to open the file for write and
can't do it, and that's the main reason that might happen.

Greg

**Comment by joecheriross** [comment]
Yes yes, the directory does exists. I checked it first. Even I tried
deleting and creating cached directory again. This started happening after
I did a pull. Let me try to download a fresh one from github and try.

Thanks,
Joe

**Comment by joecheriross** [comment]
The issue is the separator ':' is somehow not getting removed from the command line argument. In the exception message you can see that ':' is also there 

```
ERROR: java.lang.RuntimeException: Can't write to models/cached/corefpruner-onto.ser.gz:
```

This I think is a problem with the newly updated code. I run this on a mac book pro. May be it can also be a problem with the java on this machine. Please check.

**Comment by joecheriross** [comment]
Sorry. that ":" has nothing to do with the exception. It just get appended with exception message. My ignorance. When I comment the line which raises this exception it works.

**Comment by yuliagrishina** [comment]
Hi,
i have the same issue on macbook pro. @joecheriross could you tell me which line in which file you commented?
thanks.

**Comment by joecheriross** [comment]
Hi, 

   In GUtil.scala there is a save function. 

```
def save[T <: Serializable](obj: T, path: String) {
    //if (new File(path).getParentFile() != null && new File(path).getParentFile().canWrite()) {
    //  throw new RuntimeException("Can't write to " + path + "something"); 
    //}
    if (path.endsWith(".gz")) saveGz(obj, path) else saveNonGz(obj, path);
  }
```

Comment the lines which are commented here.

**Comment by yuliagrishina** [comment]
should I recompile it afterwards?

**Comment by joecheriross** [comment]
Yes, compilation is required since the source is modified.
Note the path and name of the newly compiled jar and make sure you run with
the new jar.

**Comment by yuliagrishina** [comment]
thanks, it worked.
had to change path to wiki-db-onto.ser.gz, too, because it wasn't in /cache/.

**Comment by joecheriross** [comment]
Great.

**Comment by kpich** [comment]
Hey Greg,

I think this is a bug, and `canWrite` should be negated in GUtil.save. Can submit a PR if you want/agree...

**Comment by gregdurrett** [comment]
Yep, that's definitely wrong...I must have never noticed because I write models to the current directory so the parent ends up being null.

**Comment by gregdurrett** [comment]
(A PR would be nice, thanks!)

**Comment by kpich** [comment]
woo, see PR #11
